### DISCUSSION TITLE
Hash-based CSP solution for SSG

### DISCUSSION THREAD

**Discussion by vanadu** [vercel/next.js#64554] [Help]
### Summary

Arent' we long overdue for a hash-based solution for SSG that is as well-integrated and documented as the nonce-based solution for SSR, which is documented here: [https://nextjs.org/docs/app/building-your-application/configuring/content-security-policy](url)? Is Nextjs trying to phase out SSG or is it just low on the totem pole? I wish I'd known that going in. I'm feeling like SSG has been a big waste of time.

**Comment by darthmaim** [comment]
Hash based CSP is also required for PPR, so I hope this will be implemented soon...

**Reply by vanadu** [reply]
I'd say Next/Vercel is banking on the app router and Next 14 where this issue will probably be moot. Sad...

**Reply by darthmaim** [reply]
PPR is supposed to be the future (see https://vercel.com/blog/partial-prerendering-with-next-js-creating-a-new-default-rendering-model), but doesn't support CSP, same as any static pages (both app/pages router).

**Reply by vanadu** [reply]
So does this mean there's going to be two separate CSP/security solutions for every page rendered with PPR? I don't think Next/Vercel is taking these new security protocols very seriously at all. That's not something you can just throw out there are expect the community to deal with it for you. This should be discussed on SO - maybe then they'd pay some attention. https://stackoverflow.com/questions/78320411/using-hashes-for-content-security-policy-in-statically-generated-sites-ssg-in

**Comment by gtg7784** [comment]
is there any updates?

**Comment by olabrahmi** [comment]
Any updates on this? @leerob @delbaoliveira @ijjk

**Reply by darthmaim** [reply]
Maybe this PR can be used a first step to support this:
- #73891
